### PR TITLE
App models

### PR DESCRIPTION
**Pull request by MustakimMasum**
Activities of this branch

- Added Car model

Kindly asking for a review

### CODE REVIEW

**Review by hamidazim321 — CHANGES_REQUESTED** [review]
# 🟦 Request Changes
Great work implementing CRUD applications on the car model. Everything works great, however, 1 minor change is required.

File: app/controllers/cars_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      render json: @car
+    else
+      render json: @car.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /cars/1
+  def destroy
+    @car.destroy!
+  end
```
Comment (hamidazim321) [inline app/controllers/cars_controller.rb]: Please consider for the event where the Car is not found or the user is not authorized to do so. In any case where item could not be destroyed, like you have for the other actions. Please add an `if else` block to consider the event where the action is failed

Comment (SAZZAD1Q2) [reply app/controllers/cars_controller.rb]: # Is it like this?
def destroy
  if @car.destroy
    redirect_to cars_path, notice: 'Car was successfully destroyed.'
  else
    redirect_to cars_path, alert: 'Failed to destroy the car.'
  end
end

Comment (hamidazim321) [reply app/controllers/cars_controller.rb]: No @SAZZAD1Q2, we don't redirect a user from the API. That is for server-side rendering when have views on our rails app to redirect the user to. We don't know about any URLs on API to redirect the user.

```suggestion
  # DELETE /cars/1
  def destroy
    if @car.destroy
      head :no_content
    else
      render json: { errors: @car.errors.full_messages.to_sentence }, status: :unprocessable_entity
  end
```

**Review by hamidazim321 — APPROVED** [review]
# 🥇 APPROVED
Amazing work implementing all controller actions. There's nothing else to say other than its time to merge it. Great work 👏

**Pull request merged**